File: LeetCode/1252.py
#Link: https://leetcode.com/problems/cells-with-odd-values-in-a-matrix/

class Solution:
    def oddCells(self, n: int, m: int, indices: List[List[int]]) -> int:
        rows = [False]*n
        cols = [False]*m
        count = 0
        
        for xy in indices:
            rows[xy[0]] = not rows[xy[0]]
            cols[xy[1]] = not cols[xy[1]]
        
        for i in rows:
            for j in cols:
                if i ^ j:
                    count += 1
                    
        return count
        

File: LeetCode/cpp/1252.cpp
//Link: https://leetcode.com/problems/cells-with-odd-values-in-a-matrix/

class Solution {
public:
    int oddCells(int n, int m, vector<vector<int>>& indices) {
        vector<bool> rows(n, false);
        vector<bool> cols(m, false);
        int count = 0;
        
        for(auto xy : indices) {
            rows[xy[0]] = !rows[xy[0]];
            cols[xy[1]] = !cols[xy[1]];
        }
        
        for(bool i : rows)
            for(bool j : cols)
                if(i ^ j)
                    count += 1;
        
        return count;
    }
};
